src/worker/storage-models.ts

```diff
@@ -19,6 +19,14 @@ export interface StoredWordList {
    * The actual contents of said file.
    */
   wordlist: WordList;
+  /**
+   * Size of the file in bytes.
+   */
+  size?: number;
+  /**
+   * File extension.
+   */
+  type?: string;
```

Comment:
Please make them MANDATORY!

```suggestion
  /**
   * Size of the file in bytes.
   */
  size: number;
  /**
   * File extension.
   */
  type: string;
```

You may need to write an [upgrader function](https://dexie.org/docs/Version/Version.upgrade()#sample), that flushes the database as you mentioned.

Reply:
I still have `type` as optional, but I'll be working on it right away to address it. Would that be ok with you?

Reply:
I've manually cleaned up files cached in the DB for now, where would the upgrader function go if they would be implemented and when should we call it?

Reply:
If you're going to work on #235 directly after this, then, yes, that's fine!

I just noticed that this PR doesn't actually update the database schema and it must! >.<

Here is where the schema declaration and update function would go: https://github.com/eddieantonio/predictive-text-studio/blob/764510b168f2e6eacc1ad3b3552687162b680995/src/worker/storage.ts#L91

I just realized that I misconstrued the `SCHEMA_VERSION` constant. It turns out that if you need to change how data is represented, the Dexie schema must have `this.version(...)` lines. So here's what I'm proposing:

Delete the `SCHEMA_VERSION` constant, then add each schema revision:

```typescript
    this.version(4).stores({
      /* "Initial" schema: */
      files: "++id, name, wordlist",
      projectData:   "++id, langName, bcp47Tag, authorName, modelID, copyright, version",
      keyboardData: "++id, bcp47Tag, language, timestamp",
      KMPFileData: "++id, package",
    });

    /* Version 5: Add "size"  property to file table: */
    this.version(5).stores({
      files: "++id, name, wordlist, size"
    }).upgrade((transaction) => {
      return transaction.table("files").toCollection().modify((file: StoredWordlist) => {
        file.size = file.wordlist.length
      });
    });
```

Please see this documentation: https://dexie.org/docs/Tutorial/Design#database-versioning